src/lib.rs

```diff
@@ -39,7 +39,23 @@
 mod de;
 mod error;
 mod ser;
+pub mod types;
 
-pub use crate::de::depythonize;
+pub use crate::de::{depythonize, depythonize_custom};
 pub use crate::error::{PythonizeError, Result};
-pub use crate::ser::pythonize;
+pub use crate::ser::{pythonize, pythonize_custom};
+
+use crate::types::*;
+use pyo3::types::{PyDict, PyList};
+
+pub trait Pythonize<'py> {
+    type Dict: PyDictionary<'py>;
+    type List: PyArray<'py>;
+}
+
+pub struct PythonizeDefault;
+
+impl<'py> Pythonize<'py> for PythonizeDefault {
+    type Dict = &'py PyDict;
+    type List = &'py PyList;
+}
```

Comment:
I like this trait, however I think the name could be clearer. Maybe `PythonizerTypes`, and `DefaultPythonizerTypes` for the default?